src/overlays/actors/ovl_En_Okarina_Tag/z_en_okarina_tag.c

```diff
@@ -20,21 +33,313 @@ const ActorInit En_Okarina_Tag_InitVars = {
+    // "Ocarina tag outbreak"
+    osSyncPrintf(VT_FGCOL(GREEN) "☆☆☆☆☆ オカリナタグ発生 ☆☆☆☆☆ %x\n" VT_RST, this->actor.params);
+    this->actor.flags &= ~1;
+    this->unk_150 = (this->actor.params >> 0xA) & 0x3F;
+    this->unk_152 = (this->actor.params >> 6) & 0xF;
+    this->switchFlag = this->actor.params & 0x3F;
+    if (this->switchFlag == 0x3F) {
+        this->switchFlag = -1;
+    }
+    if (this->unk_152 == 0xF) {
+        this->unk_152 = 0;
+        this->unk_158 = 1;
+    }
+    this->actor.unk_1F = 1;
+    if (this->actor.posRot.rot.z > 0) {
+        this->unk_15C = this->actor.posRot.rot.z * 40.0f;
+    }
+
+    // "Save information"
+    osSyncPrintf(VT_FGCOL(GREEN) "☆☆☆☆☆ セーブ情報\t ☆☆☆☆☆ %d\n" VT_RST, this->switchFlag);
+    // "Type index"
+    osSyncPrintf(VT_FGCOL(YELLOW) "☆☆☆☆☆ 種類インデックス ☆☆☆☆☆ %d\n" VT_RST, this->unk_150);
+    // "Correct answer information"
+    osSyncPrintf(VT_FGCOL(PURPLE) "☆☆☆☆☆ 正解情報\t ☆☆☆☆☆ %d\n" VT_RST, this->unk_152);
+    // "Range information"
+    osSyncPrintf(VT_FGCOL(CYAN) "☆☆☆☆☆ 範囲情報\t ☆☆☆☆☆ %d\n" VT_RST, this->actor.posRot.rot.z);
+    // "Processing range information"
+    osSyncPrintf(VT_FGCOL(CYAN) "☆☆☆☆☆ 処理範囲情報\t ☆☆☆☆☆ %f\n" VT_RST, this->unk_15C);
+    // "Hit?"
+    osSyncPrintf(VT_FGCOL(GREEN) "☆☆☆☆☆ 当り？\t\t ☆☆☆☆☆ %d\n" VT_RST, this->unk_158);
+    osSyncPrintf("\n\n");
+
+    if (this->switchFlag >= 0) {
+        if (Flags_GetSwitch(globalCtx, this->switchFlag) != 0) {
+            Actor_Kill(&this->actor);
+            return;
+        }
+    }
+
+    switch (this->unk_150) {
+        case 7:
+            this->actionFunc = func_80ABEF2C;
+            break;
+        case 2:
+            if (LINK_IS_ADULT) {
+                Actor_Kill(&this->actor);
+                break;
+            }
+        case 1:
+        case 4:
+        case 6:
+            this->actionFunc = func_80ABF28C;
+            break;
+        case 5:
+            // This poem is dedicated to the memory of the dearly departed members of the Royal Family.
+            this->actor.textId = 0x5021;
+            this->actionFunc = func_80ABF708;
+            break;
+        default:
+            Actor_Kill(&this->actor);
+            break;
+    }
+}
+
+void func_80ABEF2C(EnOkarinaTag* this, GlobalContext* globalCtx) {
+    Player* player;
+    u16 unk_152;
+
+    player = PLAYER;
+    this->unk_15A++;
+    if (this->switchFlag >= 0) {
+        if (Flags_GetSwitch(globalCtx, this->switchFlag) != 0) {
+            this->actor.flags &= ~1;
+            return;
+        }
+    }
+    if ((this->unk_152 != 6) || (gSaveContext.unk_F3C[0x389] != 0)) {
+        if ((s32)(player->stateFlags2 << 7) < 0) {
+            // "North! ! ! ! !"
+            osSyncPrintf(VT_FGCOL(RED) "☆☆☆☆☆ 北！！！！！ ☆☆☆☆☆ %f\n" VT_RST, this->actor.xzDistFromLink);
+        }
+        if ((this->actor.xzDistFromLink < (90.0f + this->unk_15C)) &&
+            (fabsf(player->actor.posRot.pos.y - this->actor.posRot.pos.y) < 80.0f)) {
+            {
```

Comment:
What's the purpose of the extra set of brackets here?

Reply:
Oh, must have been a mistake